helix-term/src/application.rs

```diff
@@ -1003,32 +1022,21 @@ impl Application {
                         }))
                     }
                     Ok(MethodCall::WorkspaceFolders) => {
-                        let language_server =
-                            self.editor.language_servers.get_by_id(server_id).unwrap();
-
-                        Ok(json!(language_server.workspace_folders()))
+                        Ok(json!(language_server!().workspace_folders()))
                     }
                     Ok(MethodCall::WorkspaceConfiguration(params)) => {
+                        let language_server = language_server!();
                         let result: Vec<_> = params
                             .items
                             .iter()
                             .map(|item| {
-                                let mut config = match &item.scope_uri {
-                                    Some(scope) => {
-                                        let path = scope.to_file_path().ok()?;
-                                        let doc = self.editor.document_by_path(path)?;
-                                        doc.language_config()?.config.as_ref()?
-                                    }
-                                    None => self
-                                        .editor
-                                        .language_servers
-                                        .get_by_id(server_id)
-                                        .unwrap()
-                                        .config()?,
-                                };
+                                let mut config = language_server.config()?;
                                 if let Some(section) = item.section.as_ref() {
-                                    for part in section.split('.') {
-                                        config = config.get(part)?;
+                                    // for some reason some lsps send an empty string (observed in 'vscode-eslint-language-server')
+                                    if !section.is_empty() {
+                                        for part in section.split('.') {
+                                            config = config.get(part)?;
+                                        }
                                     }
```

Comment:
Why change this? Even if it's empty it should run fine with one for loop, no?

Reply:
`"".split(.)` returns `[""]` so the loop runs once and `config.get("")` returns an error (but the server wants the entire config). Not sure if the fix should be included with this PR but I can see why it's necessary.

Reply:
> Why change this? Even if it's empty it should run fine with one for loop, no?

Well the comment above it is the reason why I did this (`vscode-eslint-language-server` otherwise had issues, I'm not sure anymore what it was, but I think it was a null pointer exception or something like that).

> Not sure if the fix should be included with this PR but I can see why it's necessary.

Yeah should've probably been its own PR, I agree. I'm not sure if it's worth it now anymore, as it's such a simple change (and you have already reviewed it).

Reply:
I would be ok with leaving it in. The spec isn't entirely clear here but I believe that vscode treats a missing field and a field containing an empty string the same. Seeing that an "official" LSP uses this I think its reasonable to ne robust here. You could split the fix out into a seperate commit if not too much trouble.

Reply:
> You could split the fix out into a seperate commit if not too much trouble.

Yeah, I can do that in the next rebase onto master

Reply:
I have split it now in its own commit: https://github.com/helix-editor/helix/commit/8f1ac545351d69b6db9b2f26bae39473c74f9fc0